backend/alembic/versions/0057_multi_notes.py

```diff
@@ -0,0 +1,159 @@
+            nullable=False,
+        ),
+        sa.Column(
+            "updated_at",
+            sa.DateTime(timezone=True),
+            server_default=sa.text("now()"),
+            nullable=False,
+        ),
+        sa.Column("rom_id", sa.Integer(), nullable=False),
+        sa.Column("user_id", sa.Integer(), nullable=False),
+        sa.ForeignKeyConstraint(["rom_id"], ["roms.id"], ondelete="CASCADE"),
+        sa.ForeignKeyConstraint(["user_id"], ["users.id"], ondelete="CASCADE"),
+        sa.PrimaryKeyConstraint("id"),
+        sa.UniqueConstraint(
+            "rom_id", "user_id", "title", name="unique_rom_user_note_title"
+        ),
+    )
+
+    # Create indexes for performance
+    op.create_index("idx_rom_notes_public", "rom_notes", ["is_public"])
+    op.create_index("idx_rom_notes_rom_user", "rom_notes", ["rom_id", "user_id"])
+
+    # Get connection for manual index creation
+    connection = op.get_bind()
+    # For MariaDB compatibility, we limit the content index length
+    connection.execute(text("CREATE INDEX idx_rom_notes_title ON rom_notes (title)"))
+    connection.execute(
+        text("CREATE INDEX idx_rom_notes_content ON rom_notes (content(100))")
+    )
+
+    # Migrate existing notes from rom_user table if the columns exist
+    connection = op.get_bind()
+
+    # Check if note columns exist in rom_user table
+    inspector = sa.inspect(connection)
+    rom_user_columns = [col["name"] for col in inspector.get_columns("rom_user")]
+
+    if "note_raw_markdown" in rom_user_columns and "note_is_public" in rom_user_columns:
+        # Migrate existing notes
+        result = connection.execute(
+            text(
+                """
+                SELECT id, rom_id, user_id, note_raw_markdown, note_is_public, updated_at
+                FROM rom_user
+                WHERE note_raw_markdown IS NOT NULL AND note_raw_markdown != ''
+            """
+            )
+        )
+
+        for row in result:
+            connection.execute(
+                text(
+                    """
+                    INSERT INTO rom_notes (title, content, is_public, tags, created_at, updated_at, rom_id, user_id)
+                    VALUES (:title, :content, :is_public, :tags, :created_at, :updated_at, :rom_id, :user_id)
+                """
+                ),
+                {
+                    "title": "My Note",
+                    "content": row.note_raw_markdown,
+                    "is_public": bool(row.note_is_public),
+                    "tags": "[]",
+                    "created_at": row.updated_at or text("now()"),
+                    "updated_at": row.updated_at or text("now()"),
+                    "rom_id": row.rom_id,
+                    "user_id": row.user_id,
+                },
+            )
+        # Remove the old note columns from rom_user table
+        op.drop_column("rom_user", "note_raw_markdown")
+        op.drop_column("rom_user", "note_is_public")
```

Comment:
maybe we can drop these in a future migration so downgrading below is easier, would that be better? 🤷🏼

Reply:
dope let's leave drop the columns for cleanliness 👍🏼